### PR TITLE
Allows "peeping" through holes in walls

### PR DESCRIPTION
**Pull request by ghost**
[content]

wallhole
In particular, ones left by penetrating bullet marks and likewise. (Removing these holes with a solder as usual will prevent looking through them)
Also adds the ability for surgical drills to create these holes, after a delay. Does not work on reinforced walls, by design. (Diamond one shown above)
:cl:
 * rscadd: Holes made in walls from either drilling or bullets can now be viewed through by clicking on them while adjacent.

### CODE REVIEW

**Comment by west3436** [comment]
now make a feature to hide the hole from anyone who hasn't examined the wall yet

**Comment by Exxion** [comment]
Long drill

**Comment by ghost** [comment]
> Long drill

Will probably replace the sprite for drilled holes eventually.

**Comment by SonixApache** [comment]
not sure how i feel about the drill thing, but the other one is a great feature

File: code/modules/mob/living/living.dm [hunk 1]
```diff
@@ -737,6 +737,13 @@ Thanks.
 	if(T != loc)
 		handle_hookchain(Dir)
 
+	if(istype(client.eye,/turf/simulated/wall))
```
Comment (DamianX) [inline code/modules/mob/living/living.dm]: This will runtime for every mob/living without a client

Comment (ghost) [reply code/modules/mob/living/living.dm]: Well noted, fixed.

**Comment by Optimism333** [comment]
Interesting idea and I like it, but a question immediately comes to mind.

Will walls with holes drilled through them no longer be airtight?

**Comment by ghost** [comment]
> Interesting idea and I like it, but a question immediately comes to mind.
> 
> Will walls with holes drilled through them no longer be airtight?

They still are, but it's an idea to consider. Maybe they should leak very tiny amounts of air?

**Comment by Optimism333** [comment]
> > Interesting idea and I like it, but a question immediately comes to mind.
> > Will walls with holes drilled through them no longer be airtight?
> 
> They still are, but it's an idea to consider. Maybe they should leak very tiny amounts of air?

It would bother me from a spess logics perspective if walls with holes are somehow still airtight but false walls are not. However, it is worth considering if it can be implemented in a way that's not too annoying to deal with from a player perspective or janky.

**Pull request merged**